### PR TITLE
3829 Queries using table entity to filter on table_name and database id

### PR DESCRIPTION
**Pull request by frankfarrell**
Some of the api queries only filter on table name where they should be filtering on table name and database id. 

Also, The current migration scripts do not successfully create the correct uniqueness constraint on sqlite. This is because doesn't allow dropping unnamed unique constraints:
https://stackoverflow.com/questions/42013265/remove-unique-constraint-on-a-column-in-sqlite-database

Original issue here: https://github.com/apache/incubator-superset/issues/3829

### CODE REVIEW

**Comment by john-bodley** [comment]
@frankfarrel would you mind adding a description to help provide more context? Also there are some linting issues you need to resolve.

**Comment by frankfarrell** [comment]
@john-bodley Apologies, both updated now.

File: superset/migrations/versions/cf50994ebd3a_tables_unique_on_table_name_and_database.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+"""tables unique on table name and database
```
Comment (john-bodley) [inline superset/migrations/versions/cf50994ebd3a_tables_unique_on_table_name_and_database.py]: @frankfarrell why is this migration required, given that it seems you're only querying the data.

**Comment by frankfarrell** [comment]
The query updates are to match the table uniqueness constraint correctly.
The migration is to put that constraint in place correctly for sqpite.

File: superset/migrations/versions/cf50994ebd3a_tables_unique_on_table_name_and_database.py (hunk 1 above)
Comment (frankfarrell) [reply superset/migrations/versions/cf50994ebd3a_tables_unique_on_table_name_and_database.py]: @john-bodley The query updates are to match the table uniqueness constraint correctly.
This migration is to put that constraint in place correctly for sqlite, the previous migrations don't manage to drop the old constraint

**Pull request closed without merging**